services/tenants/transports/tenants_http.go

```diff
@@ -0,0 +1,163 @@
+package tenantstransports
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi/v5"
+
+	"sensorbucket.nl/sensorbucket/internal/httpfilter"
+	"sensorbucket.nl/sensorbucket/internal/pagination"
+	"sensorbucket.nl/sensorbucket/internal/web"
+	"sensorbucket.nl/sensorbucket/services/tenants/tenants"
+)
+
+func NewTenantsHTTP(r chi.Router, tenantSvc tenantService, url string) *TenantsHTTPTransport {
+	t := &TenantsHTTPTransport{
+		router:    r,
+		tenantSvc: tenantSvc,
+		url:       url,
+	}
+	t.setupRoutes(t.router)
+	return t
+}
+
+type TenantsHTTPTransport struct {
+	router    chi.Router
+	tenantSvc tenantService
+	url       string
+}
+
+func (t *TenantsHTTPTransport) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	t.router.ServeHTTP(w, r)
+}
+
+func (t *TenantsHTTPTransport) setupRoutes(r chi.Router) {
+	r.Get("/tenants/list", t.httpGetTenants())
+	r.Post("/tenants", t.httpCreateTenant())
+	r.Delete("/tenants/{tenant_id}", t.httpDeleteTenant())
+}
+
+func (t *TenantsHTTPTransport) httpCreateTenant() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		dto := tenants.TenantDTO{}
+		defer r.Body.Close()
+		err := json.NewDecoder(r.Body).Decode(&dto)
+		if err != nil {
+			web.HTTPResponse(w, http.StatusBadRequest, web.APIResponseAny{
+				Message: "Invalid JSON body",
+			})
+			return
+		}
+		if validationErrors := ensureValuesNotEmptyOrZero(
+			map[string]interface{}{
+				"name":                   dto.Name,
+				"address":                dto.Address,
+				"zip_code":               dto.ZipCode,
+				"city":                   dto.City,
+				"chamber_of_commerce_id": dto.ChamberOfCommerceID,
+				"headquarter_id":         dto.HeadquarterID,
+			},
+		); len(validationErrors) > 0 {
+			web.HTTPResponse(w, http.StatusBadRequest, web.APIResponseAny{
+				Message: "model not valid",
+				Data:    validationErrors,
+			})
+			return
+		}
+		created, err := t.tenantSvc.CreateNewTenant(dto)
+		if err != nil {
+			if errors.Is(err, tenants.ErrParentTenantNotFound) {
+				web.HTTPResponse(w, http.StatusNotFound, web.APIResponseAny{
+					Message: "Parent tenant could not be found",
+				})
+				return
+			} else {
+				web.HTTPError(w, err)
+				return
+			}
```

Comment:
Web.HTTPError should be sufficient. If the error is created with `web.NewError()` it will use the correct status code and message